### PR TITLE
feat(dms/kafka): add new resources user and permissions

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

add new resources user and permissions

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add new resources user and permissions
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run TestAccDmsKafkaUser_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run TestAccDmsKafkaUser_basic -timeout 360m -parallel 4
=== RUN   TestAccDmsKafkaUser_basic
=== PAUSE TestAccDmsKafkaUser_basic
=== CONT  TestAccDmsKafkaUser_basic
--- PASS: TestAccDmsKafkaUser_basic (630.72s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       630.896s

make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run TestAccDmsKafkaPermissions_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run TestAccDmsKafkaPermissions_basic -timeout 360m -parallel 4
=== RUN   TestAccDmsKafkaPermissions_basic
=== PAUSE TestAccDmsKafkaPermissions_basic
=== CONT  TestAccDmsKafkaPermissions_basic
--- PASS: TestAccDmsKafkaPermissions_basic (728.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       728.857s
```

### CODE REVIEW

File: docs/resources/dms_kafka_user.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+* `region` - (Optional, String, ForceNew) The region in which to create the DMS kafka user resource. If omitted, the
+  provider-level region will be used. Changing this creates a new resource.
+
+* `instance_id` - (Required, String, ForceNew) Specifies the ID of the DMS kafka instance to which the user belongs.
+  Changing this creates a new resource.
+
+* `name` - (Required, String, ForceNew) Specifies the name of the user. Changing this creates a new resource.
+
+* `password` - (Required, String) Specifies the password of the user. The parameter must be 8 to 32 characters
+  long and contain only letters(case-sensitive), digits, and special characters(`~!@#$%^&*()-_=+|[{}]:'",<.>/?).
```
Comment (hmbig2) [inline docs/resources/dms_kafka_user.md:34]: add validation to schema?

Comment (Jason-Zhang9309) [reply docs/resources/dms_kafka_user.md:34]: the rule is vary loose, so validation is not necessary

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_user.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+	updateOpts := &model.ResetUserPasswrodRequest{
+		InstanceId: instanceId,
+		UserName:   d.Get("name").(string),
+		Body: &model.ResetUserPasswrodReq{
+			NewPassword: utils.String(d.Get("password").(string)),
+		},
+	}
+
+	log.Printf("[DEBUG] Update DMS instance user options: %#v", updateOpts)
```
Comment (chengxiangdong) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_user.go]: the plaintext password is printed, the password should be replaced with *** in the log

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_permissions.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/huaweicloud/huaweicloud-sdk-go-v3/services/kafka/v2/model"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
```
Comment (ShiChangkuo) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_permissions.go]: both used fmtp and fmt packages

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_permissions.go [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+	c := meta.(*config.Config)
+	client, err := c.HcDmsV2Client(c.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("error creating DMS client: %s", err)
+	}
+
+	topicName := d.Get("topic_name").(string)
+	instanceId := d.Get("instance_id").(string)
+
+	createOpts := &model.UpdateTopicAccessPolicyRequest{
```
Comment (ShiChangkuo) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_permissions.go]: it's better to change the variable name

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_user.go (hunk 2 above)
Comment (ShiChangkuo) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_user.go]: it seems that the value is a pointer

Comment (Jason-Zhang9309) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_user.go]: this log removed

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_user_test.go [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+	}
+
+	return nil, fmt.Errorf("can not found DMS kafka user")
+}
+
+func TestAccDmsKafkaUser_basic(t *testing.T) {
+	var user model.ShowInstanceUsersEntity
+	rName := acceptance.RandomAccResourceNameWithDash()
+	resourceName := "huaweicloud_dms_kafka_user.test"
+	password := "Test@1234"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_user_test.go]: can you try to use acceptance.RandomPassword

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_permissions.go (hunk 3 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_permissions.go]: fixed

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_user_test.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_user_test.go]: sure

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_permissions.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_permissions.go]: fixed

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run TestAccDmsKafkaUser_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run TestAccDmsKafkaUser_basic -timeout 360m -parallel 4
=== RUN   TestAccDmsKafkaUser_basic
=== PAUSE TestAccDmsKafkaUser_basic
=== CONT  TestAccDmsKafkaUser_basic
--- PASS: TestAccDmsKafkaUser_basic (626.94s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       627.031s

make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run TestAccDmsKafkaPermissions_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run TestAccDmsKafkaPermissions_basic -timeout 360m -parallel 4
=== RUN   TestAccDmsKafkaPermissions_basic
=== PAUSE TestAccDmsKafkaPermissions_basic
=== CONT  TestAccDmsKafkaPermissions_basic
--- PASS: TestAccDmsKafkaPermissions_basic (664.49s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       664.576s
```

**Pull request merged**